src/OrbitGgp/MockGgp/Working.cpp

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) 2021 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include <chrono>
+#include <cstring>
+#include <iostream>
+#include <thread>
+
+int GgpVersion(char* argv[]) {
+  if (strcmp(argv[1], "version") != 0) {
+    std::cout << "arguments are formatted wrong" << std::endl;
+    return 1;
+  }
+
+  std::cout << "12345.1.67.0 Mon 12 Dec 2012 12:12:12 PM UTC" << std::endl;
+  return 0;
+}
+
+int GgpInstanceList(char* argv[]) {
+  if (strcmp(argv[1], "instance") != 0 || strcmp(argv[2], "list") != 0 ||
+      strcmp(argv[3], "-s") != 0) {
+    std::cout << "arguments are formatted wrong" << std::endl;
+    return 1;
+  }
+  std::cout << "\
+[\n\
+ {\n\
+  \"displayName\": \"displayName-1\",\n\
+  \"id\": \"id/of/instance1\",\n\
+  \"ipAddress\": \"123.456.789.012\",\n\
+  \"lastUpdated\": \"2012-12-12T12:12:12Z\",\n\
+  \"owner\": \"owner@\",\n\
+  \"pool\": \"pool-of-test_instance_1\"\n\
+ },\n\
+ {\n\
+  \"displayName\": \"displayName-2\",\n\
+  \"id\": \"id/of/instance2\",\n\
+  \"ipAddress\": \"123.456.789.012\",\n\
+  \"lastUpdated\": \"2012-12-12T12:12:12Z\",\n\
+  \"owner\": \"owner@\",\n\
+  \"pool\": \"pool-of-test_instance_2\"\n\
+ }\n\
+]\n\
+";
+  return 0;
+}
+
+int GgpSshInit(char* argv[]) {
+  if (strcmp(argv[1], "ssh") != 0 || strcmp(argv[2], "init") != 0 || strcmp(argv[3], "-s") != 0 ||
+      strcmp(argv[4], "--instance") != 0 || strcmp(argv[5], "instance/test/id") != 0) {
+    std::cout << "arguments are wrong" << std::endl;
+    return 1;
+  }
+  std::cout << "\
+{\n\
+ \"host\": \"123.456.789.012\",\n\
+ \"keyPath\": \"example/path/to/a/key\",\n\
+ \"knownHostsPath\": \"example/path/to/known_hosts\",\n\
+ \"port\": \"12345\",\n\
+ \"user\": \"example_user\"\n\
+}\n\
+";
+  return 0;
+}
+
+int main(int argc, char* argv[]) {
+  std::this_thread::sleep_for(std::chrono::milliseconds{30});
```

Comment:
Could you add a comment explaining this waiting? I assume the idea is to test the timeout feature?

Reply:
done